examples/MerminMagic/main.jl

```diff
@@ -1,85 +1,81 @@
 # Mermin-Peres Magic Square Game Algorithm
 # https://www.gregegan.net/SCIENCE/MPMS/MPMS.html
 using Yao
-using Yao.EasyBuild, YaoPlots
-using Random
+
+# measurement operator table
+# see Table S1 in https://arxiv.org/abs/2206.12042
+op_mtx = [kron(I2, Z) kron(Z, I2) kron(Z, Z); kron(X, I2) kron(I2, X) kron(X, X); kron(-X, Z) kron(-Z, X) kron(Y, Y)]
+
+"""
+    bell_state(which::Int)::AbstractRegister
+
+Initialize bell state on a quantum register.
+
+# Arguments
+- 'which::Int': which of the four bell states to initialize
+  1 corresponds to ``(|00> + |11>)/√2``
+  2 corresponds to ``(|00> - |11>)/√2``
+  3 corresponds to ``(|01> + |10>)/√2``
+  4 corresponds to ``(|01> - |10>)/√2``
+"""
+function bell_state(which::Int)
+
+    @assert 1 <= which <= 4 "Input $(which) is not in range 1..4"
+    reg = zero_state(2)
+    apply!(reg, chain(put(2, 1 => H), cnot(2, 1, 2)))
+    (which == 2 || which == 4) && apply!(reg, put(2, 1 => Z))
+    (which == 3 || which == 4) && apply!(reg, put(2, 1 => X))
+    return reg
+end
 
 """
-    init_double_bell_state(reg::ArrayReg)
+    init_double_bell_state()::AbstractRegister
 
-Initialize two pairs of bell state.
+Initialize two pairs of bell state ``(|00> + |11>) ⊗ (|00> + |11>)/2``.
 
 First qubit of each bell state pair belongs to Alice and
 the others belong to Bob.
-
-# Arguments
-- 'reg::ArrayReg': the array of quantum register to initize state upon
 """
-function init_double_bell_state(reg::ArrayReg)
-    for k in [1, 3]
-        apply!(reg, put(4, k => H))
-        apply!(reg, cnot(4, k, k + 1))
-    end
+function init_double_bell_state()
```

Comment:
It seems you do not yet fully understand what functions are for. It is not only a subroutine, but also a way to abstract an operation for future reusing.

If a function is so specialized that you do not expect to reuse it in the future, please remove it. Here, if you want to "initialize two pairs of bell state", you do not need a function. The same applies for the `meas_by_idx`. The code can be much shorter if you remove these two functions. You are suffering from using functions like this.

Just view functions as a tool to avoid repeated patterns in your code, i.e. make your code DRY. So use function when you are about to input the same information twice.

I prefer `bell_state` over `init_double_bell_state` because, `bell_state` is used twice bellow:
```julia
join(bell_state(1), bell_state(1))
```
It is a well-defined and easy to understand, which means in can be reused in the future development.